src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -531,7 +529,7 @@ private static int GetFloatingPointMaxDigitsAndPrecision(char fmt, ref int preci
             }
 
             char fmt = ParseFormatSpecifier(format, out int precision);
-            byte* pDigits = stackalloc byte[DoubleNumberBufferLength];
+            byte* pDigits = stackalloc byte[DoubleNumberBufferLength];//TODO can't change to Span, or the call to NumberToString throws CS8350 and CS8352 (because of ValueStringBuilder reference)
```

Comment:
I'm trying to figure out how to workaround this issue. The compiler error is actually right, because sb lifetime is longer.
A option I was considering is to create the the ValueStringBuilder inside this method, because the caller only instantiate it and doesn't do anything else before passing as argument.
But I will need to return a Tuple<String, ValueStringBuilder> what is not pretty.

Another ugly option is to change the return type to ValueStringBuilder, and instead of return the string, append the string in the ValueStringBuilder object

Reply:
I think I found a better way. I can create the pDigits buffer in the caller.